contrib/meson.build

```diff
@@ -0,0 +1,23 @@
+libryml = static_library(
+	'ryml', ['rapidyaml/rapidyaml.hpp'],
+	include_directories : include_directories('rapidyaml')
+)
+
+ryml_dep = declare_dependency(
+	include_directories : include_directories('rapidyaml'),
+	link_with : libryml
+)
+
+librandom = static_library(
+    'random', ['random/random.cpp'],
+    include_directories : include_directories('random')
+)
+
+random_dep = declare_dependency(
+    include_directories : include_directories('random'),
+    link_with : librandom
+)
+
+cli_dep = declare_dependency(include_directories : include_directories('CLI'))
+
+doctest_dep = declare_dependency(include_directories: include_directories('doctest'))
```

Comment:
In general all of these should not be here. You're vendoring external code into your project which you then have to update, it's difficult to tell just from reading the source code where you got it from, and you cannot use linux distro packaged versions of your dependencies.

Meson has a mechanism to do all this for you "correctly": https://mesonbuild.com/Wrap-dependency-system-manual.html

To use it, do the the following:

```
mkdir -p subprojects
meson wrap install cli11
meson wrap install doctest
```

... and check the resulting `subprojects/*.wrap` files into git. They are tiny ini files which describe how to fetch the dependency.

Then, to use it you simply look up dependencies as normal: `dependency('CLI11')` will provide you everything you need to link to it, including (if it isn't installed globally on your system) downloading the source code, checksumming it, and configuring it as a subproject.

Reply:
I don't know where random.cpp comes from, rapidyaml could be trivially ported to meson and submitted to the Meson WrapDB.